### PR TITLE
Update GHA publishing steps

### PR DESCRIPTION
**Pull request by michaelyeager-wf**
## Which problem is this PR solving?

This PR updates the GHA publish steps to allow publishing this package to be automated. 

## Checklist:

- [ ] Unit tests have been added
- [ ] Documentation has been updated

### CODE REVIEW

**Comment by michaelyeager-wf** [comment]
Prior art:  https://github.com/Workiva/json_schema/blob/master/.github/workflows/publish.yaml

**Comment by aviary2-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

**Review by nickshoust-wf — APPROVED** [review]
Looks good to me based and worked for the other repo 

Im not sure if there's a way to test this prior to tag - so lets see how it goes!

Check here for latest version getting pushed from the action 
https://pub.dev/packages/opentelemetry

**Review by keruitan-wk — APPROVED** [review]

**Review by blakeroberts-wk — DISMISSED** [review]

**Review by nickshoust-wf — APPROVED** [review]

**Review by changliu-wk — APPROVED** [review]

**Comment by michaelyeager-wf** [comment]
@Workiva/release-management-p

**Comment by michaelyeager-wf** [comment]
Added steps tested working in another repo.  We cannot test here without pushing, and cannot do so before merge. 

QA +1

File: .github/workflows/publish.yaml [hunk 1]
```diff
@@ -8,8 +8,33 @@ on:
...
+      - name: Publish SBOM to Release Assets
+        uses: anchore/sbom-action@v0
+        with:
+          path: ./
+          format: cyclonedx-json
   publish:
-    permissions:
-      id-token: write # Required for authentication using OIDC
-    uses: dart-lang/setup-dart/.github/workflows/publish.yml@v1
+    name: Publish to pub.dev
```
Comment (chrisgustavsen-wf) [inline .github/workflows/publish.yaml:28]: Is there any reason here that we aren't using the supplied action? We just need to enable the package to run automated publishes and it should be good to go?

Comment (chrisgustavsen-wf) [reply .github/workflows/publish.yaml:28]: https://dart.dev/tools/pub/automated-publishing

Comment (michaelyeager-wf) [reply .github/workflows/publish.yaml:28]: >We just need to enable the package to run automated publishes and it should be good to go?

We've made the pub-side configuration updates as-described, but using the shared workflow from the linked documentation has resulted in the following in the past:
```
dart-lang/setup-dart/.github/workflows/publish.yml@v1 is not allowed to be used in Workiva/opentelemetry-dart. 
Reusable workflows in this workflow must be: 
within a repository that belongs to your Enterprise account or matching the following: 
dart-lang/setup-dart@*, peter-evans/create-pull-request@*, denoland/setup-deno@*, romeovs/lcov-reporter-action@*, dorny/test-reporter@refs/tags/v1.6.0, dorny/test-reporter@e9fa2f582c0ebbe2e263fd18fad744d52e0b0203, maxim-lobanov/setup-xcode@9a697e2b393340c3cacd97468baa318e4c883d98, pypa/gh-action-pypi-publish@*, subosito/flutter-action@*, coursier/setup-action@*, browser-actions/setup-chrome@*, mikepenz/action-junit-report@150e2f992e4fad1379da2056d1d1c279f520e058, invertase/github-action-dart-analyzer@1cda5922c6369263b1c7e2fbe281f69704f4d63e, golangci/golangci-lint-action@639cd343e1d3b897ff35927a75193d57cfcba299, mikefarah/yq-portable-yaml-processor@845d4ae38915c31cc76ab968393657cceaec60fc, Workiva/gha-integration-testing@26fa4acd141cb143661af9e...
```
as here:  https://github.com/Workiva/opentelemetry-dart/actions/runs/6828348643

**Comment by michaelyeager-wf** [comment]
Closing to open a fresh PR for compliance with audit tooling.

**Pull request closed without merging**